### PR TITLE
return directly without ifs in remaining packages

### PR DESCRIPTION
**Pull request by unclejack**
**- What I did**
I've modified the code which wasn't modified by #29366 and #29422 to return directly where possible. There were few such issues left to be able to break up the changes made by this PR into smaller PRs.

**- Description for the changelog**
return directly without ifs in remaining packages

### CODE REVIEW

File: integration-cli/docker_cli_external_volume_driver_unix_test.go [hunk 1]
```diff
@@ -103,10 +103,8 @@ func newVolumePlugin(c *check.C, name string) *volumePlugin {
 	read := func(b io.ReadCloser) (pluginRequest, error) {
 		defer b.Close()
 		var pr pluginRequest
-		if err := json.NewDecoder(b).Decode(&pr); err != nil {
-			return pr, err
-		}
-		return pr, nil
+		err := json.NewDecoder(b).Decode(&pr)
+		return pr, err
```
Comment (duglin) [inline integration-cli/docker_cli_external_volume_driver_unix_test.go]: if you really want to do the work for the compiler you can make it:
`return pr, json.NewDecoder(b).Decode(&pr)`  :-)

**Comment by duglin** [comment]
LGTM if janky is ok

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
I'll wait with merge, because there are some PR's running for 1.13, and I don't want to risk a merge conflict. We can merge after

**Comment by unclejack** [comment]
@thaJeztah: The other PRs are more important and this is why I've split these changes into multiple smaller PRs.  This one can wait as it's not a big deal to rebase.

**Comment by thaJeztah** [comment]
let me assign @vieux so that he can press the button if he thinks it's good to go

**Comment by thaJeztah** [comment]
RC-4 is done; let's merge!

Thanks @unclejack !

**Pull request merged**